### PR TITLE
[Test] Add tests for Stability

### PR DESCRIPTION
**Pull request by shantanutrip**
*GitHub Issue #, if available:*

**Note**: 
- If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

- All PR's are checked weekly for staleness. This PR will be closed if not updated in 30 days.

### Description
Add tests for Stability

### Tests run

**NOTE: By default, docker builds are disabled. In order to build your container, please update dlc_developer_config.toml and specify the framework to build in "build_frameworks"**
- [ ] I have run builds/tests on commit <INSERT COMMIT ID> for my changes.

**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

### Formatting
- [ ] I have run `black -l 100` on my code (formatting tool: https://black.readthedocs.io/en/stable/getting_started.html)

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by arjkesh** [comment]
LGTM, I like this way better as well for SM tests. Would you also mind running pytorch SM tests to make sure that stabilityai gets skipped?

File: test/sagemaker_tests/pytorch/inference/integration/sagemaker/test_stabilityai.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+def test_mnist_distributed_gpu_stabilityai_two(
+    framework_version, ecr_image, instance_type, sagemaker_regions
+):
+    raise Exception("Deliberate error 2")
+
+
+@pytest.mark.model("mnist")
+@pytest.mark.processor("gpu")
+@pytest.mark.gpu_test
+def test_mnist_distributed_gpu_stabilityai_three(
```
Comment (arjkesh) [inline test/sagemaker_tests/pytorch/inference/integration/sagemaker/test_stabilityai.py:37]: this one needs stability_ai_only marker, it is currently executing on PT tests

File: test/sagemaker_tests/pytorch/inference/integration/sagemaker/test_stabilityai.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+def test_mnist_distributed_gpu_stabilityai_one(
+    framework_version, ecr_image, instance_type, sagemaker_regions
+):
+    raise Exception("Deliberate error 1")
+
+
+@pytest.mark.model("mnist")
+@pytest.mark.processor("gpu")
+@pytest.mark.gpu_test
+@pytest.mark.skip_stabilityai
```
Comment (arjkesh) [inline test/sagemaker_tests/pytorch/inference/integration/sagemaker/test_stabilityai.py:27]: This should be stability_only as well

File: test/sagemaker_tests/pytorch/inference/integration/sagemaker/test_stabilityai.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+
+@pytest.mark.model("mnist")
+@pytest.mark.processor("gpu")
+@pytest.mark.gpu_test
+@pytest.mark.stabilityai_only
+def test_mnist_distributed_gpu_stabilityai_one(
+    framework_version, ecr_image, instance_type, sagemaker_regions
+):
+    raise Exception("Deliberate error 1")
```
Comment (arjkesh) [inline test/sagemaker_tests/pytorch/inference/integration/sagemaker/test_stabilityai.py:21]: Are these just for testing?

File: test/sagemaker_tests/pytorch/inference/integration/sagemaker/test_stabilityai.py (hunk 1 above)
Comment (shantanutrip) [reply test/sagemaker_tests/pytorch/inference/integration/sagemaker/test_stabilityai.py:37]: Yes, I deliberately left it for this commit. Will add the `stability_only` marker in the next commit. Thanks!

File: test/sagemaker_tests/pytorch/inference/integration/sagemaker/test_stabilityai.py (hunk 3 above)
Comment (shantanutrip) [reply test/sagemaker_tests/pytorch/inference/integration/sagemaker/test_stabilityai.py:21]: Yes, these are just for testing.

File: test/sagemaker_tests/pytorch/inference/integration/sagemaker/test_stabilityai.py (hunk 2 above)
Comment (shantanutrip) [reply test/sagemaker_tests/pytorch/inference/integration/sagemaker/test_stabilityai.py:27]: Yes, I deliberately left it for this commit. Will add the `stability_only` marker in the next commit. Thanks!

**Review by arjkesh — APPROVED** [review]

**Pull request merged**